packages/compass-saved-aggregations-queries/src/components/saved-item-card.tsx

```diff
@@ -0,0 +1,260 @@
+import React, { useCallback, useEffect, useRef, useState } from 'react';
+import {
+  Card,
+  css,
+  useDefaultAction,
+  Badge,
+  Icon,
+  Subtitle,
+  spacing,
+  uiColors,
+  useFocusState,
+  useHoverState,
+  mergeProps,
+  FocusState,
+  Menu,
+  MenuItem,
+  IconButton,
+} from '@mongodb-js/compass-components';
+import { formatDate } from '../util/format-date';
+
+type Action = 'open' | 'delete' | 'copy' | 'edit';
```

Comment:
will there ever be an 'edit' or is just rename? Very loosely related .. did we think about renaming in place?  cc @Sgrinfy

Reply:
Any editing will happen in the aggregation builder. The aggregation can only be renamed from the card view.

Reply:
Got it, I'll rename this to `rename`